cli/src/main/java/com/devonfw/tools/ide/tool/ToolCommandlet.java

```diff
@@ -94,23 +95,42 @@ public void run() {
    * @param processMode see {@link ProcessMode}
    * @param toolVersion the explicit version (pattern) to run. Typically {@code null} to ensure the configured version is installed and use that one.
    *     Otherwise, the specified version will be installed in the software repository without touching and IDE installation and used to run.
+   * @param existsEnvironmentContext the {@link Boolean} that checks if an environment context exits. If true, then the process context is passed to the
+   *     installation method as an argument.
    * @param args the command-line arguments to run the tool.
    */
-  public void runTool(ProcessMode processMode, VersionIdentifier toolVersion, String... args) {
+  public void runTool(ProcessMode processMode, VersionIdentifier toolVersion, boolean existsEnvironmentContext, String... args) {
 
     Path binaryPath;
-    Path toolPath = Path.of(getBinaryName());
+    binaryPath = Path.of(getBinaryName());
+    ProcessContext pc;
+
     if (toolVersion == null) {
-      install(true);
-      binaryPath = toolPath;
+      if (existsEnvironmentContext) {
+        pc = this.context.newProcess().errorHandling(ProcessErrorHandling.THROW).executable(binaryPath).addArgs(args);
+        install(pc, true);
+      } else {
+        install(true);
+        pc = this.context.newProcess().errorHandling(ProcessErrorHandling.THROW).executable(binaryPath).addArgs(args);
+      }
```

Comment:
Could you avoid the code duplication?
```suggestion
      pc = this.context.newProcess().errorHandling(ProcessErrorHandling.THROW).executable(binaryPath).addArgs(args);
      if (existsEnvironmentContext) {
        install(pc, true);
      } else {
        install(true);
      }
```

Reply:
The problem here is that if I do the suggested change, many tests of tools stop running. As far as I've debugged, this happens because the pc must be initialized after the install method call for these tools. But for tomcat ( at least until now it is only tomcat ) the pc must be initialized before the call because it should be passed as an argument to the method call, namely install(pc, true); So I couldn't find another solution for this problem, but to duplicate that line of code

Reply:
> The problem here is that if I do the suggested change, many tests of tools stop running.

So what was the error then? We need to understand the problems to come up with a clean design.

Reply:
As discussed, I created a private method for this case, but it seems we have found a bug in the flow of the program.